optuna/study/study.py

```diff
@@ -329,7 +329,7 @@ def objective(trial):
                 The number of trials for each process. If this argument is set to :obj:`None`,
                 there is no limitation on the number of trials. If ``timeout`` is also set to
                 :obj:`None`, the study continues to create trials until it receives a termination
-                signal such as Ctrl+C or SIGTERM.
+                signal such as Ctrl+C or SIGTERM, in which case it also kills the current trial.
```

Comment:
The current phrase "it also kills the current trial" sounds a bit ambigious to me (some users would not understand what the word "kill" means exactly). 
How about rewriting this sentence or adding a phrase such as "i.e., set the state of the trail to TrialState.FAILED"?

Reply:
Besides, I think that this PR can handle Ctrl+C (SIGINT) but cannot SIGTERM. So, the current doc seems a bit misleading in this aspect. This is just an idea, but we can replace "in which case" with "if the signal is SIGINT (Ctrl+C)" to make the condition clear.

Reply:
SIGTERM could probably also be handled, but it would require more work with signal handling, and furthermore probably wouldn't be compatible with windows. So that was not in the scope of my PR